File: ccs/backend/app/api/endpoints/small_box_info.py
"""
FastAPI 路由
自动生成由 ccs-generate
"""

from fastapi import APIRouter, Depends, HTTPException
from sqlalchemy.orm import Session
from typing import List
from app.core.database import get_db
from app.schemas.small_box_info import SmallBoxInfoResponse, SmallBoxInfoCreate, SmallBoxInfoUpdate
from app.services.small_box_info import SmallBoxInfoService

router = APIRouter(prefix="/small_box_info", tags=["small_box_info"])


@router.get("/", response_model=List[SmallBoxInfoResponse])
def list_small_box_info(skip: int = 0, limit: int = 100, db: Session = Depends(get_db)):
    """获取小箱状况列表"""
    return SmallBoxInfoService.get_all(db, skip, limit)


@router.get("/{id}", response_model=SmallBoxInfoResponse)
def get_small_box_info(id: int, db: Session = Depends(get_db)):
    """根据ID获取小箱状况"""
    obj = SmallBoxInfoService.get_by_id(db, id)
    if not obj:
        raise HTTPException(status_code=404, detail="SmallBoxInfo not found")
    return obj


@router.post("/", response_model=SmallBoxInfoResponse)
def create_small_box_info(data: SmallBoxInfoCreate, db: Session = Depends(get_db)):
    """创建小箱状况"""
    return SmallBoxInfoService.create(db, data)


@router.put("/{id}", response_model=SmallBoxInfoResponse)
def update_small_box_info(id: int, data: SmallBoxInfoUpdate, db: Session = Depends(get_db)):
    """更新小箱状况"""
    obj = SmallBoxInfoService.update(db, id, data)
    if not obj:
        raise HTTPException(status_code=404, detail="SmallBoxInfo not found")
    return obj


@router.delete("/{id}")
def delete_small_box_info(id: int, db: Session = Depends(get_db)):
    """删除小箱状况"""
    success = SmallBoxInfoService.delete(db, id)
    if not success:
        raise HTTPException(status_code=404, detail="SmallBoxInfo not found")
    return {"success": True}

File: ccs/frontend/src/api/small_box_info.ts
/**
 * API 客户端
 * 自动生成由 ccs-generate
 */

import { SmallBoxInfo, SmallBoxInfoCreate, SmallBoxInfoUpdate } from '../types/small_box_info';

const API_BASE = process.env.NEXT_PUBLIC_API_URL || 'http://localhost:8000/api';

/**
 * 获取列表
 */
export async function getSmallBoxInfoList(skip: number = 0, limit: number = 100): Promise<SmallBoxInfo[]> {
  const response = await fetch(`${API_BASE}/small_box_info/?skip=${skip}&limit=${limit}`);
  if (!response.ok) {
    throw new Error('Failed to fetch small_box_info list');
  }
  return response.json();
}

/**
 * 根据ID获取
 */
export async function getSmallBoxInfoById(id: number): Promise<SmallBoxInfo> {
  const response = await fetch(`${API_BASE}/small_box_info/${id}`);
  if (!response.ok) {
    throw new Error('Failed to fetch small_box_info');
  }
  return response.json();
}

/**
 * 创建
 */
export async function createSmallBoxInfo(data: SmallBoxInfoCreate): Promise<SmallBoxInfo> {
  const response = await fetch(`${API_BASE}/small_box_info/`, {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    body: JSON.stringify(data),
  });
  if (!response.ok) {
    throw new Error('Failed to create small_box_info');
  }
  return response.json();
}

/**
 * 更新
 */
export async function updateSmallBoxInfo(id: number, data: SmallBoxInfoUpdate): Promise<SmallBoxInfo> {
  const response = await fetch(`${API_BASE}/small_box_info/${id}`, {
    method: 'PUT',
    headers: {
      'Content-Type': 'application/json',
    },
    body: JSON.stringify(data),
  });
  if (!response.ok) {
    throw new Error('Failed to update small_box_info');
  }
  return response.json();
}

/**
 * 删除
 */
export async function deleteSmallBoxInfo(id: number): Promise<void> {
  const response = await fetch(`${API_BASE}/small_box_info/${id}`, {
    method: 'DELETE',
  });
  if (!response.ok) {
    throw new Error('Failed to delete small_box_info');
  }
}
